File: illchess-game-service/illchess-game/illchess-game-domain/src/main/java/pl/illchess/game/domain/game/exception/GameNotFoundException.java
package pl.illchess.game.domain.game.exception;

import pl.illchess.game.domain.game.model.GameId;
import pl.illchess.game.domain.commons.exception.DomainException;

import java.util.UUID;

public class GameNotFoundException extends DomainException {
    public GameNotFoundException(GameId gameId) {
        super("Game with given id = %s not found".formatted(gameId.uuid()));
    }

    public GameNotFoundException(UUID gameId) {
        super("Game with given id = %s not found".formatted(gameId));
    }
}

File: illchess-player-info-service/illchess-player-info/illchess-player-info-domain/src/main/kotlin/pl/illchess/player_info/domain/game/exception/GameNotFoundException.kt
package pl.illchess.player_info.domain.game.exception

import pl.illchess.player_info.domain.commons.exception.DomainException
import pl.illchess.player_info.domain.game.model.GameId

class GameNotFoundException(gameId: GameId) : DomainException("Game with given id ${gameId.uuid} not found")